docs/dev-client-networking.md

```diff
@@ -3,259 +3,647 @@ id: dev-client-networking
+functions or additional semantics to cover these cases.
 
-By default, `network instantiated` entities are removed when its `creator` (a client) disconnects. In order to persist these entities (and its descendants) the entity must be `pinned`. Only `network instantiated` entities can be `pinned`.
 
-To `pin` an `network instantiated` entity, a client calls `createEntityState` . This will save the current state of the entity (and its descendants) to the database. We say that the entity (and its descendants) are `persistent`.
+### Async initialization
 
-To update the state of a persistent entity, a client calls `updateEntityState`.
+A critical property of the networked system that enables the async
+initialization to work is that descendants of `networked instantiated`
+entities are assigned network IDs deterministically, even in cases where
+some parts of a descendant hierarchy fails to load. This ensures that the
+descendants can load in any order (or even fail to load) without causing a
+client to delete, overwrite, or ignore descendant updates.
 
-To delete the saved entity state of a persistent entity, a client calls `deleteEntityState`. Note that deleting the saved entity state is not the same as deleting the entity. It simply means that the information saved to the database about this entity will be deleted.
 
-When the client connects to a hub channel, it calls `listEntityStates` in order to receive the entity states that have been saved to the database.
+### Event handlers
 
-Saved entity states include `CreateMessage` s for the `network instantiated` entity and `UpdateMessage` s for itself and its descendants. These messages are queued and later processed by the `networkReceiveSystem` like any other messages. The client&rsquo;s eventually consistent properties guarantee that if entity state updates that come from the database are out-of-date, they will be appropriately handled (i.e. ignored).
+Event handlers that queue messages for later processing can be found in
+`listenForNetworkMessages` defined in
+[`src/utils/listen-for-network-message.ts`](https://github.com/mozilla/hubs/blob/master/src/utils/listen-for-network-messages.ts).
 
 
-<a id="orge3d34eb"></a>
+### Persisting Networked Entity State
 
-# Networked Entity Hierarchies
+By default, `networked instanciated entities` which are created with
+[`createNetworkedEntity()`](#createnetworkedentity), are removed when their
+creator (client) is disconnects. In order to persist these entities the entity
+must be [`pinned`](https://hubs.mozilla.com/docs/hubs-features.html). Only
+`networked instanciated entities` can be `pinned`.
 
-When `createNetworkedEntity` is called, a `network instantiated` entity is created synchronously. That is, any asynchronous loading that an entity needs to do to be &ldquo;fully realized&rdquo; will happen later.
+To `pin` an `network instantiated entities`, a client calls the built-in
+`createEntityState()` function. This will save the current state of the

```

Comment:
This is more of a comment about the naming of functions but the word “persistent” seems essential in understanding what these functions do.